### ISSUE TITLE
wdio config fails on selenium/standalone-firefox docker image

### ISSUE THREAD

**Issue by freeman-g** [webdriverio/webdriverio#1852]
## The problem

I'm trying to extend the selenium/standalone-firefox docker image a little bit to run tests via webdriverio, but I get errors using wdio config.

## Environment

* WebdriverIO version: `v4.6.2`
* Node.js version: `v4.2.6`
* [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html):  `wdio testrunner`
* if wdio testrunner, running synchronous or asynchronous tests:  `didn't make it that far`
* Additional wdio packages used (if applicable):

## Details

Here's the initial setup I used before running wdio config.  (My Dockerfile)

```
FROM selenium/standalone-firefox

RUN sudo mkdir /webdriverio
WORKDIR /webdriverio

RUN sudo apt-get update && sudo apt-get install -y \
	curl \
	nodejs \
	npm

RUN sudo curl -L https://github.com/mozilla/geckodriver/releases/download/v0.11.1/geckodriver-v0.11.1-linux64.tar.gz | sudo tar xz

#Node is called nodejs on Ubuntu so it needs to be symlinked to use the name 'node'
RUN sudo ln -s /usr/bin/nodejs /usr/bin/node
RUN sudo npm install webdriverio

#Needed for to avoid errors with the display
ENV DISPLAY=:1

CMD Xvfb :1 -screen 0 1024x768x16 & java -jar -Dwebdriver.gecko.driver=./geckodriver /opt/selenium/selenium-server-standalone.jar
```

Then running wdio config fails with the following errors.  I have tried running it as root as well, but get the same errors.  The standalone-firefox image is based on Ubuntu 16.04.

```
seluser@91de4387d143:/webdriverio$ sudo su -
root@91de4387d143:~# cd /webdriverio/
root@91de4387d143:/webdriverio# ls
geckodriver  node_modules
root@91de4387d143:/webdriverio# ./node_modules/.bin/wdio config

=========================
WDIO Configuration Helper
=========================

? Where do you want to execute your tests? On my local machine
? Which framework do you want to use? mocha
? Shall I install the framework adapter for you? Yes
? Where are your test specs located? ./test/specs/**/*.js
? Which reporter do you want to use?  dot - https://github.com/webdriverio/wdio-dot-reporter
? Shall I install the reporter library for you? Yes
? Do you want to add a service to your test setup? (Press <space> to select, <a> to toggle all, <i> to inverse selection)
? Level of logging verbosity silent
? In which directory should screenshots gets saved if a command fails? ./errorShots/
? What is the base url? http://localhost

Installing wdio packages:
pkg: wdio-mocha-framework
pkg: wdio-dot-reporter
/webdriverio/node_modules/webdriverio/build/lib/cli.js:278
                    throw err;
                    ^

Error: Command failed: /bin/sh -c npm i -D wdio-mocha-framework wdio-dot-reporter
npm WARN enoent ENOENT: no such file or directory, open '/webdriverio/package.json'
npm WARN webdriverio No description
npm WARN webdriverio No repository field.
npm WARN webdriverio No README data
npm WARN webdriverio No license field.
npm ERR! Linux 4.4.41-boot2docker
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "i" "-D" "wdio-mocha-framework" "wdio-dot-reporter"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! path /webdriverio/node_modules/wdio-dot-reporter
npm ERR! code EXDEV
npm ERR! errno -18
npm ERR! syscall rename

npm ERR! EXDEV: cross-device link not permitted, rename '/webdriverio/node_modules/wdio-dot-reporter' -> '/webdriverio/node_modules/.wdio-dot-reporter.DELETE'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /webdriverio/npm-debug.log

    at ChildProcess.exithandler (child_process.js:213:12)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:821:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

**Comment by christian-bromann** [comment]
This is not an issue with WebdriverIO. Try to install the `wdio-dot-reporter` package manually. Seems that the docker container doesn't allow to install npm packages. Either `npm-install-package` faults or an issue with the configuration of the container.

**Comment by freeman-g** [comment]
Thanks for responding.  I had installed webdriverio itself via npm succcessfully in the Dockerfile.  Then I was testing some more things and trying to `npm install wdio-dot-reporter` directly in the container.  This didn't work and got the cross-device link error.

So I went back and installed wdio-dot-reporter and wdio-mocha-framework in the Dockerfile and this works:

```
RUN sudo npm install webdriverio wdio-dot-reporter wdio-mocha-framework
```
